control/controlclient/direct.go

```diff
@@ -1643,6 +1643,50 @@ func (c *Direct) ReportHealthChange(w *health.Warnable, us *health.UnhealthyStat
 	res.Body.Close()
 }
 
+// SetDeviceAttrs does a synchronous call to the control plane to update
+// the node's attributes.
+//
+// See docs on [tailcfg.SetDeviceAttributesRequest] for background.
+func (c *Auto) SetDeviceAttrs(ctx context.Context, attrs tailcfg.AttrUpdate) error {
+	return c.direct.SetDeviceAttrs(ctx, attrs)
+}
+
+// SetDeviceAttrs does a synchronous call to the control plane to update
+// the node's attributes.
+//
+// See docs on [tailcfg.SetDeviceAttributesRequest] for background.
+func (c *Direct) SetDeviceAttrs(ctx context.Context, attrs tailcfg.AttrUpdate) error {
+	np, err := c.getNoiseClient()
```

Comment:
I am curious why we don't call `DoNoiseRequest` here, which already checks for `panicOnUse`, and seems to [use the same transport](https://github.com/tailscale/tailscale/blob/30d3e7b2429bc6e2226c365dd4e92f211e09147f/control/controlclient/noise.go#L165).

This comment implies that using `getConn` directly (like `doWithBody` does) would attach node key proofs, but I don't think that's actually true? I only see `KeyProvingNoiseRoundTripper` [used](https://github.com/tailscale/tailscale/blob/30d3e7b2429bc6e2226c365dd4e92f211e09147f/tsnet/tsnet.go#L908) in tsnet.

https://github.com/tailscale/tailscale/blob/30d3e7b2429bc6e2226c365dd4e92f211e09147f/control/controlclient/noise.go#L37-L41

I am likely missing something, since it's my first time taking a closer look at noise client, but I would like to understand it better.

Reply:
Oh, right. That was added for one caller ages ago that needed to do it from afar, but all the existing callers (who used the helpers that automatically JSON-encode the request body) were never updated to use it, because the exported one isn't as helpful.

But then apparently a handful of people have since moved on to using DoNoiseRequest directly, and doing the JSON encoding themselves. But those callers aren't setting the load balancer hint header:

```
func addLBHeader(req *http.Request, nodeKey key.NodePublic) {
	if !nodeKey.IsZero() {
		req.Header.Add(tailcfg.LBHeader, nodeKey.String())
	}
}
```

Which makes the load balancer work a bit harder when it's absent.

There's about a 50/50 split between the two styles right now and I don't want to clean them up one way or another in this PR, but I do want to clean it up in its own change later.

I'll keep this as is for now (as it calls `addLBHeader`) but yeah, I don't like the split.